File: C#/Array & String/1431. Kids With the Greatest Number of Candies.cs
// 1431. Kids With the Greatest Number of Candies
// Difficulty: Easy
// Submissions Link: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/submissions/1017414992/
public class Solution {
    public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
        int maxCandies = candies.Max();
        List<bool> result = new List<bool>();
        
        foreach (int candyCount in candies) {
            result.Add(candyCount + extraCandies >= maxCandies);
        }
        
        return result;
    }
}

File: C/Array & String/1431. Kids With the Greatest Number of Candies.c
// 1431. Kids With the Greatest Number of Candies
// Difficulty: Easy
// Submission Link: https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/submissions/1017413901

bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize){
    int maxCandies = 0;
    for (int i = 0; i < candiesSize; i++) {
        if (candies[i] > maxCandies) {
            maxCandies = candies[i];
        }
    }
    
    *returnSize = candiesSize;
    bool* result = (bool*)malloc(candiesSize * sizeof(bool));
    for (int i = 0; i < candiesSize; i++) {
        result[i] = (candies[i] + extraCandies >= maxCandies);
    }
    
    return result;
}